File: lesson2.py
#!/usr/bin/python3
#
import sys
import json
import datetime
import pymysql
import pymysql.cursors
import os
import time
#
_db_user = "ipott_user"
_db_password = "ipott_pass"
_db_name = "ipott_db"
_pattern = "lesson2.html"
#
def menene():
    #
    add_new_dates()
    #
    reqmet = os.environ["REQUEST_METHOD"]
    #reqmet = "GET"
    #reqmet = "POST"
    if(reqmet == "GET"):
        get()
    elif(reqmet == "POST"):
        post()
    else:
        print("Content-type: text/html;charset=utf-8")
        print()
        print("error")
    #
#
def get():
    #
    print("Content-type: text/html;charset=utf-8")
    print()
    lakebo = get_file_contents(_pattern)
    #
    lavero = get_insert_block()
    denede = get_delete_block()
    benede = get_datalists()
    #
    lakebo = lakebo.replace(
            "<!--datalists-->",
            benede
        )
    #
    lakebo = lakebo.replace(
            "<!--insert_block-->",
            lavero
        )
    #
    '''
    lakebo = lakebo.replace(
            "<!--delete_block-->",
            denede
        )
    '''
    #
    print(lakebo)
    #
#
def post():
    #
    #print("Content-type: text/html;charset=utf-8")
    print("Content-Type: application/json;charset=utf-8")
    print()
    #
    resope = json.load(sys.stdin)
    '''
    resope = {
        "command": "insert",
        "lesson_id": "0",
        "day_id": "1",
        "lapse_id": "1",
        "subject_id": "1",
        "cabinet_id": "1",
        "teacher_id": "1",
        "class_id": "1",
    }
    '''
    #
    #
    if(False):
        print()
    elif(resope["command"] == "select"):
        #valusa = get_select_valusa(resope)
        re = select_from_lesson(resope)
        print(re)
    elif(resope["command"] == "insert"):
        valusa = get_insert_valusa(resope)
        re = insert_into_lesson(valusa)
        print(re)
    elif(resope["command"] == "update"):
        valusa = get_update_valusa(resope)
        re = update_lesson(valusa,resope["lesson_id"])
        print(re)
    elif(resope["command"] == "delete"):
        valusa = get_delete_valusa(resope)
        re = delete_from_lesson(valusa)
        print(re)
    else:
        print(resope["command"])
    #
#
def add_new_dates():
    #
    #
    connection = pymysql.connect(host='localhost',
        user=_db_user,
        password=_db_password,
        database=_db_name,
        charset='utf8mb4',
        cursorclass=pymysql.cursors.DictCursor)
    #
    cursor = connection.cursor()
    #
    dale = get_near_days(7)
    #print(f"<script>console.log('{dale}');</script>")
    #
    for he in dale:
        try:
            sql = f'INSERT INTO day VALUES (0,"{he}");'
            cursor.execute(sql)
        except Exception as ke:
            #print(ke)
            time.sleep(0)
    #
    connection.commit()
    #
    cursor.close()
    connection.close()
    #
#
def get_near_days(nu):
    #
    re = []
    #
    dt = datetime.date.today()
    dts = dt.strftime("%Y.%m.%d")
    re.append(dts)
    #
    for i in range(nu):
        dt += datetime.timedelta(days=1)
        dts = dt.strftime("%Y.%m.%d")
        re.append(dts)
    #
    return(re)
    #
#
def get_info_dict_list():
    #
    connection = pymysql.connect(host='localhost',
        user=_db_user,
        password=_db_password,
        database=_db_name,
        charset='utf8mb4',
        cursorclass=pymysql.cursors.DictCursor)
    #
    cursor = connection.cursor()
    #
    sql = ""
    sql += "SELECT "
    sql += "lesson.id AS lesson_id, "
    sql += "lesson.day_id AS day_id, "
    sql += "lesson.class_id AS class_id, "
    sql += "lesson.lapse_id AS lapse_id, "
    sql += "lesson.subject_id AS subject_id, "
    sql += "lesson.teacher_id AS teacher_id, "
    sql += "lesson.cabinet_id AS cabinet_id, "
    #
    sql += "day.name AS day_name, "
    sql += "class.short_name AS class_name, "
    sql += "lapse.n AS lapse_name, "
    sql += "subject.short_name AS subject_name, "
    sql += "teacher.short_name AS teacher_name, "
    sql += "cabinet.short_name AS cabinet_name "
    #
    sql += "FROM "
    sql += "lesson,day,class,lapse,subject,teacher,cabinet "
    sql += "WHERE "
    sql += "lesson.day_id = day.id AND "
    sql += "lesson.class_id = class.id AND "
    sql += "lesson.lapse_id = lapse.id AND "
    sql += "lesson.subject_id = subject.id AND "
    sql += "lesson.teacher_id = teacher.id AND "
    sql += "lesson.cabinet_id = cabinet.id "
    sql += "ORDER BY lapse.n"
    sql += ";"
    #
    #print(sql)
    #sql = "SELECT * FROM lesson;"
    #
    cursor.execute(sql)
    ru = cursor.fetchall()
    #print(ru)
    #
    #cursor.close()
    #connection.close()
    #
    return(ru)
    #return(sql)
#
def filter_dict_list(info_dict_list,resope):
    #
    hotu = []
    ru = info_dict_list
    re = resope
    #
    #print("filter")
    #
    if(len(ru) == 0):
      print("lo")
      return(hotu)
    #
    n = 0
    del re["command"]
    #
    #print(len(ru))
    #print(set(re.keys()))
    #
    while(n < len(ru)):
        ka = 0
        for ve in set(re.keys()):
            #
            if(str(ru[n][ve]) == str(re[ve])):
                ka+=1
            #
        #
        if(ka == len(re.keys())):
            hotu.append(ru[n])
        #
        n+=1
    #
    return(hotu)
    #
#
def select_from_lesson(resope):
    #
    hedelu = get_info_dict_list()
    #
    fedelu = filter_dict_list(hedelu,resope)
    for he in fedelu:
        he["command"] = "append"
    #
    #print(fedelu)
    #print()
    #print("sele")
    #print()
    #
    re = json.dumps(fedelu)
    #
    return(re)
    #
#
def insert_into_lesson(valusa):
    #
    connection = pymysql.connect(host='localhost',
        user=_db_user,
        password=_db_password,
        database=_db_name,
        charset='utf8mb4',
        cursorclass=pymysql.cursors.DictCursor)
    #
    cursor = connection.cursor()
    #
    sql = f"INSERT INTO lesson VALUES ({valusa});"
    cursor.execute(sql)
    #
    connection.commit()
    #
    cursor.close()
    connection.close()
    #
    rule = []
    ru = {}
    ru["command"] = "alert"
    ru["message"] = "inserting done!"
    rule.append(ru)
    re = json.dumps(rule)
    return(re)
#
def update_lesson(valusa,lesson_id):
    #
    connection = pymysql.connect(host='localhost',
        user=_db_user,
        password=_db_password,
        database=_db_name,
        charset='utf8mb4',
        cursorclass=pymysql.cursors.DictCursor)
    #
    cursor = connection.cursor()
    #
    sql = f"UPDATE lesson SET {valusa} WHERE id={lesson_id};"
    cursor.execute(sql)
    #
    connection.commit()
    #
    cursor.close()
    connection.close()
    #
    rule = []
    ru = {}
    ru["command"] = "alert"
    ru["message"] = "updating done!"
    rule.append(ru)
    re = json.dumps(rule)
    return(re)
#
def delete_from_lesson(valusa):
    #
    connection = pymysql.connect(host='localhost',
        user=_db_user,
        password=_db_password,
        database=_db_name,
        charset='utf8mb4',
        cursorclass=pymysql.cursors.DictCursor)
    #
    cursor = connection.cursor()
    #
    sql = f"DELETE FROM lesson WHERE {valusa};"
    cursor.execute(sql)
    #
    connection.commit()
    #
    cursor.close()
    connection.close()
    #
    rule = []
    ru = {}
    ru["command"] = "alert"
    ru["message"] = "deleting done!"
    rule.append(ru)
    re = json.dumps(rule)
    return(re)
#
def get_select_valusa(resope):
    #
    valusa = None
    z = 0
    va = []
    #
    if("day_id" in set(resope.keys())):
      va.append('lesson.day_id="'+resope["day_id"]+'"')
      z+=1
    if("class_id" in set(resope.keys())):
      va.append('lesson.class_id="'+resope["class_id"]+'"')
      z+=1
    if("lapse_id" in set(resope.keys())):
      va.append('lesson.lapse_id="'+resope["lapse_id"]+'"')
      z+=1
    if("subject_id" in set(resope.keys())):
      va.append('lesson.subject_id="'+resope["subject_id"]+'"')
      z+=1
    if("cabinet_id" in set(resope.keys())):
      va.append('lesson.cabinet_id="'+resope["cabinet_id"]+'"')
      z+=1
    if("teacher_id" in set(resope.keys())):
      va.append('lesson.teacher_id="'+resope["teacher_id"]+'"')
      z+=1
    #
    if(z > 0):
      valusa = " AND ".join(va)
    else:
      valusa = "lesson.id != 0"
    #
    return(valusa)
#
def get_insert_valusa(resope):
    #
    valusa = ""
    valusa += '"'+resope["lesson_id"]+'",'
    valusa += '"'+resope["day_id"]+'",'
    valusa += '"'+resope["lapse_id"]+'",'
    valusa += '"'+resope["subject_id"]+'",'
    valusa += '"'+resope["cabinet_id"]+'",'
    valusa += '"'+resope["teacher_id"]+'",'
    valusa += '"'+resope["class_id"]+'"'
    #
    return(valusa)
#
def get_update_valusa(resope):
    #
    valusa = ""
    valusa += 'day_id="'+resope["day_id"]+'",'
    valusa += 'lapse_id="'+resope["lapse_id"]+'",'
    valusa += 'subject_id="'+resope["subject_id"]+'",'
    valusa += 'cabinet_id="'+resope["cabinet_id"]+'",'
    valusa += 'teacher_id="'+resope["teacher_id"]+'",'
    valusa += 'class_id="'+resope["class_id"]+'"'
    #
    return(valusa)
#
def get_delete_valusa(resope):
    #
    valusa = ""
    valusa += 'id='+resope["lesson_id"]+''
    #
    return(valusa)
#
def get_dict_from_table(tname):
    #
    connection = pymysql.connect(host='localhost',
        user=_db_user,
        password=_db_password,
        database=_db_name,
        charset='utf8mb4',
        cursorclass=pymysql.cursors.DictCursor)
    #
    cursor = connection.cursor()
    #
    sql = f"SELECT * FROM {tname};"
    cursor.execute(sql)
    result = cursor.fetchall()
    #
    cursor.close()
    connection.close()
    #
    return(result)
    #
#
def get_search_input(tname,phold,cla,disabled=False):
    #
    leseke = ""
    leseke += f'<input type="search" list="{tname}" '
    leseke += f'placeholder="{phold}" class="{cla}" '
    leseke += 'disabled' if disabled else ''
    leseke += '>'
    #
    return(leseke)
#
def get_hidden_input(value,cla):
    #
    leseke = ""
    leseke += f'<input type="hidden" '
    leseke += f'value="{value}" class="{cla}" '
    leseke += '>'
    #
    return(leseke)
#
def get_options(tname):
    #
    tdict = get_dict_from_table(tname)
    leseke = ""
    for rova in tdict:
        rid = rova["id"]
        rnm = ""
        #
        if(False):
            rnm = ""
        elif(tname=="day"):
            rnm = rova["name"]
        elif(tname=="lapse"):
            rnm = rova["n"]
        elif(True):
            rnm = rova["short_name"]
        #
        leseke += f'<option data-id="{rid}">{rnm}</option>'
    #
    return(leseke)
    #
#
def get_datalist(tname):
    #
    leseke = ""
    leseke += f'<datalist id="{tname}">'
    leseke += get_options(tname);
    leseke += '</datalist>'
    #
    return(leseke)
    #
#
def get_datalists():
    #
    tnames = [
        "day",
        "class",
        "lapse",
        "subject",
        "teacher",
        "cabinet"
    ]
    #
    lesene = ""
    for tname in tnames:
        lesene += get_datalist(tname)
    #
    return(lesene)
#
def get_button(title,cla):
    #
    le = ""
    le += f'<button class="{cla}">{title}</button>'
    #
    return(le)
#
def get_select(hopote,kale):
    #
    kalese = " ".join(kale)
    le = ""
    le += f'<select class="{kalese}">'
    le += hopote
    le += '</select>'
    #
    return(le)
#
def get_date_input(kale):
    #
    kalese = " ".join(kale)
    le = ""
    le += f'<input type="date" class="{kalese}">'
    #
    return(le)
#
def get_insert_block():
    #
    cla1 = "select"
    cla2 = "insert"
    #
    clali = []
    clali.append(cla1)
    clali.append(cla2)
    #
    clall = " ".join(clali)
    #
    le = ""
    #
    kale = list(clali)
    kale.insert(0,"day")
    le += get_date_input(kale)
    #
    tnames = [
        "class",
        "lapse",
        "subject",
        "teacher",
        "cabinet"
    ]
    #
    lesene = ""
    for tname in tnames:
        kale = list(clali)
        kale.insert(0,tname)
        hopote = get_options(kale[0])
        le += get_select(hopote,kale)
    #
    le += get_button("Найти",cla1)
    le += get_button("Добавить",cla2)
    #
    return(le)
#
def get_delete_block():
    #
    cla1 = "update"
    cla2 = "delete"
    #
    clali = []
    clali.append(cla1)
    clali.append(cla2)
    #
    clall = " ".join(clali)
    #
    #
    le = ""
    le += get_hidden_input("6",clall)
    le += get_search_input("day","дата",clall)
    le += get_search_input("class","класс",clall)
    le += get_search_input("lapse","номер урока",clall)
    le += get_search_input("subject","предмет",clall)
    le += get_search_input("teacher","учитель",clall)
    le += get_search_input("cabinet","кабинет",clall)
    le += get_button("Изменить",cla1)
    le += get_button("Удалить",cla2)
    #
    return(le)
#
def get_file_contents(fn):
    #
    fl = open(fn,"r",encoding="utf-8")
    st = fl.read()
    fl.close()
    #
    return(st)
#
if(__name__=="__main__"):
    menene()

File: lesson2.js
async function send_post_query (
  hobode,
  sereve,
  funuke
) {
  //
  let resope = await fetch(sereve, {
    method: "POST",
    headers: {
      "Content-Type": "application/json;charset=utf-8"
    },
    body: JSON.stringify(hobode)
  });
  let resule = await resope.text();
  funuke(resule);
}
//
function parse_response(resope) {
  //
  let append_was = false;
  let hobo = JSON.parse(resope);
  //
  for(let ho of hobo) {
    if(false) {
      console.log();
    } else if(ho.command == "alert") {
      //alert(ho.message);
      hanema();
    } else if(ho.command == "append") {
      append_was = true;
      console.log("append");
      append_ho(ho);
      hanema();
    }
    console.log(ho);
  }
  //
  if(append_was) {
    append_cloneho();
  }
  //
}
//
function sleep(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}
//
function hanema() {
  let ko = document.body.classList;
  ko.toggle("hanema");
  sleep(5000).then(() => { ko.toggle("hanema"); });
}
//
function append_cloneho() {
  let cloneday_input = document.createElement('input');
  cloneday_input.setAttribute("type","date");
  cloneday_input.classList.add("day");
  cloneday_input.classList.add("clone");
  //
  let clone_button = document.createElement('button');
  clone_button.classList.add("clone");
  clone_button.addEventListener("click",clone_button_click);
  clone_button.innerHTML = "Клонировать";
  //
  let rovo_div = document.createElement('div');
  rovo_div.classList.add("rovo");
  rovo_div.classList.add("last");
  rovo_div.append(cloneday_input);
  rovo_div.append(clone_button);
  //
  let sc = document.body.querySelector("script");
  let rovo_in = document.body.insertBefore(rovo_div,sc);
  //
}
function append_ho(ho) {
  //
  let tname = null;
  let dale = null;
  //
  let hidden_input = document.createElement('input');
  hidden_input.setAttribute("type","hidden");
  hidden_input.classList.add("update");
  hidden_input.classList.add("delete");
  hidden_input.setAttribute("value",ho.lesson_id);
  //
  let day_input = document.createElement('input');
  day_input.setAttribute("type","date");
  day_input.classList.add("day");
  day_input.classList.add("update");
  day_input.classList.add("delete");
  let ho_re = ho.day_name;
  ho_re = ho_re.replace(/[.]/g,"-");
  day_input.setAttribute("value",ho_re);
  //
  tname = "class";
  let class_input = document.createElement('select');
  class_input.classList.add(tname);
  class_input.classList.add("update");
  class_input.classList.add("delete");
  dale = document.querySelector("datalist#"+tname);
  class_input.innerHTML = dale.innerHTML;
  class_input.value = ho.class_name;
  //
  tname = "lapse";
  let lapse_input = document.createElement('select');
  lapse_input.classList.add(tname);
  lapse_input.classList.add("update");
  lapse_input.classList.add("delete");
  dale = document.querySelector("datalist#"+tname);
  lapse_input.innerHTML = dale.innerHTML;
  lapse_input.value = ho.lapse_name;
  //
  tname = "subject";
  let subject_input = document.createElement('select');
  subject_input.classList.add(tname);
  subject_input.classList.add("update");
  subject_input.classList.add("delete");
  dale = document.querySelector("datalist#"+tname);
  subject_input.innerHTML = dale.innerHTML;
  subject_input.value = ho.subject_name;
  //
  tname = "teacher";
  let teacher_input = document.createElement('select');
  teacher_input.classList.add(tname);
  teacher_input.classList.add("update");
  teacher_input.classList.add("delete");
  dale = document.querySelector("datalist#"+tname);
  teacher_input.innerHTML = dale.innerHTML;
  teacher_input.value = ho.teacher_name;
  //
  tname = "cabinet";
  let cabinet_input = document.createElement('select');
  cabinet_input.classList.add(tname);
  cabinet_input.classList.add("update");
  cabinet_input.classList.add("delete");
  dale = document.querySelector("datalist#"+tname);
  cabinet_input.innerHTML = dale.innerHTML;
  cabinet_input.value = ho.cabinet_name;
  //
  let update_button = document.createElement('button');
  update_button.classList.add("update");
  update_button.addEventListener("click",update_button_click);
  update_button.innerHTML = "Изменить";
  //
  let delete_button = document.createElement('button');
  delete_button.classList.add("delete");
  delete_button.addEventListener("click",delete_button_click);
  delete_button.innerHTML = "Удалить";
  //
  let rovo_div = document.createElement('div');
  rovo_div.classList.add("rovo");
  rovo_div.append(hidden_input);
  rovo_div.append(day_input);
  rovo_div.append(class_input);
  rovo_div.append(lapse_input);
  rovo_div.append(subject_input);
  rovo_div.append(teacher_input);
  rovo_div.append(cabinet_input);
  rovo_div.append(update_button);
  rovo_div.append(delete_button);
  //
  let sc = document.body.querySelector("script");
  let rovo_in = document.body.insertBefore(rovo_div,sc);
  //
}
//
function get_value_of_hidden_input(pa) {
  //
  let seleko = "input[type='hidden']";
  let daluheva = pa.querySelector(seleko).value;
  //
  return(daluheva);
}
//
function get_value_of_search_input(daluhe,pa) {
  //
  let seleko = "input[list='"+daluhe+"']";
  let daluheva = pa.querySelector(seleko).value;
  let daluheha = null;
  //
  seleko = "#"+daluhe+" option";
  let hopotezu = document.querySelectorAll(seleko);
  //
  for(const hopotene of hopotezu) {
    if(hopotene.value == daluheva) {
      daluheha = hopotene.dataset.id;
      break;
    }
  }
  //
  return(daluheha);
  //
}
//
function get_value_of_date_input(daluhe,pa) {
  //
  let seleko = "."+daluhe;
  let daluheva = pa.querySelector(seleko).value;
  daluheva = daluheva.replace(/-/g,".");
  //
  //console.log(daluheva);
  //
  let daluheha = null;
  //
  seleko = "#"+daluhe+" option";
  let hopotezu = document.querySelectorAll(seleko);
  //
  for(const hopotene of hopotezu) {
    if(hopotene.value == daluheva) {
      daluheha = hopotene.dataset.id;
      break;
    }
  }
  //
  /*
  if(daluheha == null) {
    alert("Даты "+daluheva+" нет в базе данных!");
  }
  */
  return(daluheha);
  //
}
//
function get_value_of_select(daluhe,pa) {
  //
  let seleko = "select."+daluhe;
  let daluheva = pa.querySelector(seleko);
  let daluheme = daluheva.options[daluheva.selectedIndex];
  let daluheha = daluheme.dataset.id;
  //
  //console.log(daluheme.text);
  //console.log(daluheha);
  //
  if(daluheha == 0) {
    return(null);
  } else {
    return(daluheha);
  }
}
//
function delete_excess_rovos() {
  let rovos = document.querySelectorAll(".rovo");
  for(let rovo of rovos) {
    if(!rovo.classList.contains("first")) {
      rovo.remove();
    }
  }
}
//
function select_button_click(event) {
  let button = event.currentTarget;
  let pa = button.parentNode;
  let mapa = new Map();
  //
  //alert("hi");
  let ka = "select";
  //alert(ka);
  //
  let command = ka;
  let day_id = get_value_of_date_input("day",pa);
  let class_id = get_value_of_select("class",pa);
  let lapse_id = get_value_of_select("lapse",pa);
  let subject_id = get_value_of_select("subject",pa);
  let teacher_id = get_value_of_select("teacher",pa);
  let cabinet_id = get_value_of_select("cabinet",pa);
  //
  let z = 0;
  mapa.set("command",command);
  if(day_id!=null){mapa.set("day_id",day_id);z++;}
  if(class_id!=null){mapa.set("class_id",class_id);z++;}
  if(lapse_id!=null){mapa.set("lapse_id",lapse_id);z++;}
  if(subject_id!=null){mapa.set("subject_id",subject_id);z++;}
  if(teacher_id!=null){mapa.set("teacher_id",teacher_id);z++;}
  if(cabinet_id!=null){mapa.set("cabinet_id",cabinet_id);z++;}
  //
  if(z > 0) {
    let hobo = Object.fromEntries(mapa);
    //
    console.log(hobo);
    //
    send_post_query(hobo,"/ipott/lesson.py",parse_response);
  }
  //
  delete_excess_rovos();
}
//
function pepe_hobo(rovo) {
  //
  let pa = rovo;
  //
  let mapa = new Map();
  let hobo = null;
  //
  //alert("hi");
  let ka = "insert";
  //alert(ka);
  //
  let command = ka;
  let lesson_id = "0";
  let day_id = get_value_of_date_input("day",pa);
  let class_id = get_value_of_select("class",pa);
  let lapse_id = get_value_of_select("lapse",pa);
  let subject_id = get_value_of_select("subject",pa);
  let teacher_id = get_value_of_select("teacher",pa);
  let cabinet_id = get_value_of_select("cabinet",pa);
  //
  let z = 0;
  mapa.set("command",command);
  mapa.set("lesson_id",lesson_id);
  if(day_id!=null){mapa.set("day_id",day_id);z++;}
  if(class_id!=null){mapa.set("class_id",class_id);z++;}
  if(lapse_id!=null){mapa.set("lapse_id",lapse_id);z++;}
  if(subject_id!=null){mapa.set("subject_id",subject_id);z++;}
  if(teacher_id!=null){mapa.set("teacher_id",teacher_id);z++;}
  if(cabinet_id!=null){mapa.set("cabinet_id",cabinet_id);z++;}
  //
  if(z == 6) {
    hobo = Object.fromEntries(mapa);
  }
  return(hobo);
}
//
function se_hobo(hobo) {
  //
  if(hobo === null) {
    //
    alert("Одно из полей заполнено неверно!");
    //
  } else {
    //
    console.log(hobo);
    //
    send_post_query(hobo,"/ipott/lesson.py",parse_response);
    //
  }
  //
}
//
function insert_button_click(event) {
  let button = event.currentTarget;
  let rovo = button.parentNode;
  //
  let hobo = pepe_hobo(rovo);
  se_hobo(hobo);
  //
}
//
function update_button_click(event) {
  let button = event.currentTarget;
  let pa = button.parentNode;
  let mapa = new Map();
  //
  //alert("hi");
  let ka = "update";
  //alert(ka);
  //
  let command = ka;
  let lesson_id = get_value_of_hidden_input(pa);
  let day_id = get_value_of_date_input("day",pa);
  let class_id = get_value_of_select("class",pa);
  let lapse_id = get_value_of_select("lapse",pa);
  let subject_id = get_value_of_select("subject",pa);
  let teacher_id = get_value_of_select("teacher",pa);
  let cabinet_id = get_value_of_select("cabinet",pa);
  //
  let z = 0;
  mapa.set("command",command);
  mapa.set("lesson_id",lesson_id);
  if(day_id!=null){mapa.set("day_id",day_id);z++;}
  if(class_id!=null){mapa.set("class_id",class_id);z++;}
  if(lapse_id!=null){mapa.set("lapse_id",lapse_id);z++;}
  if(subject_id!=null){mapa.set("subject_id",subject_id);z++;}
  if(teacher_id!=null){mapa.set("teacher_id",teacher_id);z++;}
  if(cabinet_id!=null){mapa.set("cabinet_id",cabinet_id);z++;}
  //
  if(z == 6) {
    let hobo = Object.fromEntries(mapa);
    //
    console.log(hobo);
    //
    send_post_query(hobo,"/ipott/lesson.py",parse_response);
  }
}
//
function delete_button_click(event) {
  let button = event.currentTarget;
  let pa = button.parentNode;
  let mapa = new Map();
  //
  //alert("hi");
  let ka = "delete";
  //alert(ka);
  //
  let command = ka;
  let lesson_id = get_value_of_hidden_input(pa);
  //
  mapa.set("command",command);
  mapa.set("lesson_id",lesson_id);
  //
  let hobo = Object.fromEntries(mapa);
  //
  console.log(hobo);
  //
  send_post_query(hobo,"/ipott/lesson.py",parse_response);
  //
  pa.remove();
}
function clone_button_click(event) {
  //
  let button = event.currentTarget;
  let pa = button.parentNode;
  //
  let rovos = document.querySelectorAll(".rovo");
  //
  let cloneday_id = get_value_of_date_input("day",pa);
  //
  for(let rovo of rovos) {
    if(!rovo.classList.contains("first")) {
      if(!rovo.classList.contains("last")) {
        //
        let mapa = new Map();
        //
        let hobo = pepe_hobo(rovo);
        //
        hobo.day_id = cloneday_id;
        //
        //console.log(hobo);
        //
        se_hobo(hobo);
        //
      }
    }
  }
}
//
/*
function insert_day_blur() {
  //
  let ka = "insert";
  //
  let wday = document.querySelector("input[list='weekday']."+ka);
  //
  let day = document.querySelector("input[list='day']."+ka);
  day = day.value;
  //
  let daypa = day.split(".");
  let now = new Date(daypa[0],daypa[1]-1,daypa[2]);
  nade = now.getDay();
  //
  seleko = "#weekday option";
  let hopotezu = document.querySelectorAll(seleko);
  //
  for(const hopotene of hopotezu) {
    if(hopotene.dataset.n == nade) {
      wday.value = hopotene.value;
      break;
    }
  }
}
*/
//
/*
function delete_day_blur() {
  //
  let ka = "delete";
  //
  let wday = document.querySelector("input[list='weekday']."+ka);
  //
  let day = document.querySelector("input[list='day']."+ka);
  day = day.value;
  //
  let daypa = day.split(".");
  let now = new Date(daypa[0],daypa[1]-1,daypa[2]);
  nade = now.getDay();
  //
  seleko = "#weekday option";
  let hopotezu = document.querySelectorAll(seleko);
  //
  for(const hopotene of hopotezu) {
    if(hopotene.dataset.n == nade) {
      wday.value = hopotene.value;
      break;
    }
  }
}
*/
//
let ins_buttons = document.querySelectorAll("button.insert");
for(const ins_button of ins_buttons) {
  ins_button.addEventListener("click",insert_button_click);
}
//
let del_buttons = document.querySelectorAll("button.delete");
for(const del_button of del_buttons) {
  del_button.addEventListener("click",delete_button_click);
}
//
let upd_buttons = document.querySelectorAll("button.update");
for(const upd_button of upd_buttons) {
  upd_button.addEventListener("click",update_button_click);
}
//
let sel_buttons = document.querySelectorAll("button.select");
for(const sel_button of sel_buttons) {
  sel_button.addEventListener("click",select_button_click);
}
//
/*
let i_day = document.querySelector("input[list='day'].insert");
i_day.addEventListener("blur",insert_day_blur);
*/
//
/*
let d_day = document.querySelector("input[list='day'].delete");
d_day.addEventListener("blur",delete_day_blur);
*/
//
